libminifi/include/core/ClassLoader.h

```diff
@@ -56,182 +47,6 @@ namespace core {
-    return nullptr;
-  }
-
-  /**
-   * Returns an initializer for the factory.
-   */
-  virtual std::unique_ptr<ObjectFactoryInitializer> getInitializer() {
-    return nullptr;
-  }
-
-  /**
-   * Gets the name of the object.
-   * @return class name of processor
-   */
-  virtual std::string getName() = 0;
-
-  virtual std::string getGroupName() const {
-    return group_;
-  }
-
-  virtual std::string getClassName() = 0;
-  /**
-   * Gets the class name for the object
-   * @return class name for the processor.
-   */
-  virtual std::vector<std::string> getClassNames() = 0;
-
-  virtual std::unique_ptr<ObjectFactory> assign(const std::string &class_name) = 0;
-
- private:
-  std::string group_;
-};
-/**
- * Factory that is used as an interface for
- * creating processors from shared objects.
- */
-template<class T>
-class DefautObjectFactory : public ObjectFactory {
- public:
-  DefautObjectFactory() {
-    className = core::getClassName<T>();
-  }
-
-  DefautObjectFactory(const std::string &group_name) // NOLINT
-      : ObjectFactory(group_name) {
-    className = core::getClassName<T>();
-  }
-  /**
-   * Virtual destructor.
-   */
-  virtual ~DefautObjectFactory() = default;
-
-  /**
-   * Create a shared pointer to a new processor.
-   */
-  virtual std::shared_ptr<CoreComponent> create(const std::string &name) {
-    std::shared_ptr<T> ptr = std::make_shared<T>(name);
-    return std::static_pointer_cast<CoreComponent>(ptr);
-  }
-
-  /**
-   * Create a shared pointer to a new processor.
-   */
-  virtual std::shared_ptr<CoreComponent> create(const std::string &name, const utils::Identifier &uuid) {
-    std::shared_ptr<T> ptr = std::make_shared<T>(name, uuid);
-    return std::static_pointer_cast<CoreComponent>(ptr);
-  }
-
-  /**
-   * Create a shared pointer to a new processor.
-   */
-  virtual CoreComponent* createRaw(const std::string &name) {
-    T *ptr = new T(name);
-    return dynamic_cast<CoreComponent*>(ptr);
-  }
-
-  /**
-   * Create a shared pointer to a new processor.
-   */
-  virtual CoreComponent* createRaw(const std::string &name, const utils::Identifier &uuid) {
-    T *ptr = new T(name, uuid);
-    return dynamic_cast<CoreComponent*>(ptr);
-  }
-
-  /**
-   * Gets the name of the object.
-   * @return class name of processor
-   */
-  virtual std::string getName() {
-    return className;
-  }
-
-  /**
-   * Gets the class name for the object
-   * @return class name for the processor.
-   */
-  virtual std::string getClassName() {
-    return className;
-  }
-
-  virtual std::vector<std::string> getClassNames() {
-    std::vector<std::string> container;
-    container.push_back(className);
-    return container;
-  }
-
-  virtual std::unique_ptr<ObjectFactory> assign(const std::string& /*class_name*/) {
-    return nullptr;
-  }
-
- protected:
-  std::string className;
-};
-
 /**
  * Function that is used to create the
  * processor factory from the shared object.
```

Comment:
the `createFactory` typedef is no longer used, it can be removed

Reply:
removed